lib/src/iterable_extensions.dart

```diff
@@ -549,6 +550,54 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return true;
   }
+
+  /// Returns an iterable whose elements are contiguous slices of [this].
+  ///
+  /// Each slice is [length] elements long, except for the last one which may be
+  /// shorter if [this] contains too few elements. Each slice begins after the
+  /// last one ends.
+  ///
+  /// For example, `(1, 2, 3, 4, 5).slices(2)` returns `((1, 2), (3, 4), (5))`.
+  Iterable<Iterable<T>> slices(int length) sync* {
+    if (length < 1) throw RangeError.range(length, 1, null, 'length');
+
+    var iterator = this.iterator;
+    while (iterator.moveNext()) {
+      yield () sync* {
+        var i = 0;
+        do {
+          yield iterator.current;
+          i++;
+        } while (i < length && iterator.moveNext());
+      }();
+    }
+  }
+
+  /// Returns an iterable of each subsequence of [this] of the given [length].
+  ///
+  /// Each subsequence is exactly [length] elements long and shares `length - 1`
+  /// elements with the previous subsequence. Returns an empty iterable if
+  /// [length] is greater than this iterable's length.
+  ///
+  /// For example, `(1, 2, 3, 4).subsequences(2)` returns
+  /// `([1, 2], [2, 3], [3, 4])`.
+  Iterable<List<T>> subsequences(int length) sync* {
+    if (length < 1) throw RangeError.range(length, 1, null, 'length');
+
+    var iterator = this.iterator;
+    var subsequence = <T>[];
+    while (iterator.moveNext()) {
```

Comment:
> I'd probably split this into two loops, instead of branching in the main loop on something which is going to always be false up to some point, then always true after. It really is two loops.

Done.

> Maybe:
> 
> ```dart
> var buffer = ListQueue<T>(length + 1);
> while (buffer.length < length) {
>   if (iterator.moveNext()) {
>     buffer.add(iterator.current);
>   } else {
>     return;
>   }
> }
> while (true) {
>   yield List<T>.unmodifiable(buffer);
>   if (!iterator.moveNext()) return;
>   buffer..removeFirst()..add(iterator.current);
> }
> ```

I don't think the `ListQueue` actually makes this more efficient. You're still copying `length` elements on every iteration in the `List.unmodifiable()` call.